### PR TITLE
feat: introduce a dedicated namespace for OTPs and SPPs

### PR DESCRIPTION
**Pull request by srieteja**
**- What I did**
- introduce a dedicated namespace for OTPs and SPPs
- Provide backwards compatibility for legacy OTP and SPP keys

**- How I did it**
- Introduced a class variable in OtpVerbHandler called that has a value of '__otp'; which will be the namespace for all OTPs and SPPs
- Allow backward compatibility: check for OTP and SPP without the otp namespace.

**- How to verify it**
- Unit tests added that validate that OTPs and SPPs that use legacy style keys are also considered valid
- functional/e2e tests will be once this PR is merged into https://github.com/atsign-foundation/at_server/pull/1955

**- Description for the changelog**

feat: introduce a dedicated namespace for OTPs and SPPs

### CODE REVIEW

File: packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart [hunk 1]
```diff
@@ -257,23 +259,46 @@ abstract class AbstractVerbHandler implements VerbHandler {
...
+      logger.finest('No SPP found in KeyStore. Validating as OTP');
     }
+    if (sppAtData != null &&
+        sppAtData.data?.toLowerCase() == otp.toLowerCase()) {
+      return true;
+    }
+
     // If SPP is not valid, then check if the provided otp is valid.
-    String otpKey =
+    String otpLegacyKey =
```
Comment (sitaram-kalluri) [inline packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart]: Remove "otpLegacyKey" from the keystore after the use.

**Review by sitaram-kalluri — CHANGES_REQUESTED** [review]
Remove "otpLegacyKey" from the keystore after the use.

File: packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart (hunk 1 above)
Comment (srieteja) [reply packages/at_secondary_server/lib/src/verb/handler/abstract_verb_handler.dart]: @sitaram-kalluri As we discussed offline, I have refactored the if blocks to use only one variable and one keystore fetch. This way at the end whichever otp key was used to validate/store otp will be removed with the code in [d1bcd37](https://github.com/atsign-foundation/at_server/pull/1956/commits/d1bcd379d62acae5e1ae6f50a70e15e3698e01ce)

File: packages/at_secondary_server/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ## 3.0.46
 - fix: Default OTP expiry value remains unchanged for the subsequent "otp:" requests
-
+- feat: OTPs stored with a dedicated namespace
```
Comment (sitaram-kalluri) [inline packages/at_secondary_server/CHANGELOG.md]: Suggest to please rewrite  as "Introduced a dedicated namespace for storing OTPs."

**Review by sitaram-kalluri — DISMISSED** [review]
Suggest to rewrite the line in changelog.md, Otherwise LGTM

File: packages/at_secondary_server/CHANGELOG.md (hunk 2 above)
Comment (srieteja) [reply packages/at_secondary_server/CHANGELOG.md]: changed as part of [f4505d5](https://github.com/atsign-foundation/at_server/pull/1956/commits/f4505d55ed9284d8ab5d3a13ba7c5bedf7f8154d)

**Review by gkc — APPROVED** [review]
Merged conflicts from trunk into this branch, also did a few small cleanup changes

LGTM but @murali-shris @sitaram-kalluri can one of you review and merge please

**Comment by srieteja** [comment]
Thank you @gkc

**Pull request merged**